src/jit/hwintrinsiclistxarch.h

```diff
@@ -11,58 +11,58 @@
 // clang-format off
 
 #if FEATURE_HW_INTRINSICS
-//                  Intrinsic ID                   Function name                ISA   
-//  SSE Intrinsics          
-HARDWARE_INTRINSIC(SSE_IsSupported,             "get_IsSupported",              SSE)
-HARDWARE_INTRINSIC(SSE_Add,                     "Add",                          SSE)
+//                 Intrinsic ID                         Function name                   ISA             Instructions (flt, dbl, i8, i16, i32, i64)                                          ival
```

Comment:
IMO we should support 128 bit operands (some insertions and extractions work on them) and make a reserved value/values for future use (binary16 et al IEEE 754:2008)

Reply:
Additionally there are conversions between base types: explicit and implicit i.e. unpack../pack.. etc.

Reply:
Perhaps it could be useful to differentiate between mem/reg/imm operands already here - just a general idea.

Reply:
128-bit operands, if we have a way to emit constants for them (and I'm not sure if we do right now), can be handled based off a single ssize_t variable (all fields are duplicated).

I can clarify, but the instruction encoding is made to be keyed off the baseType, not the targetType.

I don't think differentiating between mem/reg/imm here is needed. The instruction emitted should be the same and it can be handled in codegen.